### ISSUE TITLE
无法监听普通的网络请求

### ISSUE THREAD

**Issue by TabCen** [HDB-Li/LLDebugTool#34]
在项目中可监听到图片的请求、以及UMeng的一些请求，其他请求不可监听

**Comment by HDB-Li** [comment]
其他的请求是怎么调用的   是AFURLSession还是CFNetwork

**Comment by qcyl** [comment]
是否是使用moya+rxswift发送的请求呢？我这边使用moya+rxswift发送的请求是无法监听到的

**Comment by HDB-Li** [comment]
Thanks,  I will try it. @qcyl

**Comment by qcyl** [comment]
你可以用我的demo试一下 https://github.com/qcyl/rxswift_yiwei

**Comment by HDB-Li** [comment]
@qcyl  Thanks!! 😆

**Comment by HDB-Li** [comment]
@qcyl   Thanks, I think it's enough to deal the session created with configuration by hook `[NSURLSessionConfiguration defaultSessionConfiguration]` and `[ephemeralSessionConfiguration]`, maybe I need hook this two methods too. 

```
+ (NSURLSession *)sessionWithConfiguration:(NSURLSessionConfiguration *)configuration;
+ (NSURLSession *)sessionWithConfiguration:(NSURLSessionConfiguration *)configuration delegate:(nullable id <NSURLSessionDelegate>)delegate delegateQueue:(nullable NSOperationQueue *)queue;
``

Thanks again.

**Comment by HDB-Li** [comment]
@qcyl  I tried hook protocolClasses in [NSURLSessionConfiguration load] like this, but method5 is nil. I'm not sure why I can't hook the get method of the attribute during the load.
```
    Method method5 = class_getInstanceMethod([NSURLSessionConfiguration class], @selector(protocolClasses));
    Method method6 = class_getInstanceMethod([NSURLSessionConfiguration class], @selector(LL_protocolClasses));
    method_exchangeImplementations(method5, method6);
```

So I think to deal this problem I need hook this.
```
+ (NSURLSession *)sessionWithConfiguration:(NSURLSessionConfiguration *)configuration;
+ (NSURLSession *)sessionWithConfiguration:(NSURLSessionConfiguration *)configuration delegate:(nullable id <NSURLSessionDelegate>)delegate delegateQueue:(nullable NSOperationQueue *)queue;
```

**Comment by qcyl** [comment]
你可以看一下这篇文章[https://juejin.im/entry/58ed8c6344d904005772e8c7](https://juejin.im/entry/58ed8c6344d904005772e8c7)
他hook的class是``Class cls =NSClassFromString(@"__NSCFURLSessionConfiguration")?:NSClassFromString(@"NSURLSessionConfiguration");``
我试了是可以拦截到请求的，但是会打印下面的东西，我不太清楚这些是什么意思
![image](https://user-images.githubusercontent.com/15316136/45813558-8cd42680-bd05-11e8-8a1b-ceaeba8dbbcc.png)

**Comment by HDB-Li** [comment]
Thanks, I was hook the `NSURLSessionConfiguration`, I will try `__NSCFURLSessionConfiguration` tomorrow.

The log in screenshot means you init a UIView in sub thread.  Is It called by LLDebugTool?

**Comment by HDB-Li** [comment]
In screenshot. It was called in [Config initia], but I didn't init UIView in this method.  Did you modify this file?

**Comment by qcyl** [comment]
并没有，我只在``LLURLProtocol.m``文件中加了下面三个方法
```
+ (void)load {
  Class cls =NSClassFromString(@"__NSCFURLSessionConfiguration")?:NSClassFromString(@"NSURLSessionConfiguration");
  [self swizzleSelector:@selector(protocolClasses) fromClass:cls toClass:[self class]];
}
+ (void)swizzleSelector:(SEL)selector fromClass:(Class)original toClass:(Class)stub {
  Method originalMethod =class_getInstanceMethod(original,selector);
  Method stubMethod =class_getInstanceMethod(stub,selector);
  if (!originalMethod ||!stubMethod){
    [NSException raise:NSInternalInconsistencyException format:@"Couldn't load NEURLSessionConfiguration."];
  }
  method_exchangeImplementations(originalMethod,stubMethod);
}
-(NSArray*)protocolClasses{
  return @[[LLURLProtocol class]];
}
```

**Comment by HDB-Li** [comment]
Hi @qcyl,  You are right, 😆   I can't hook `NSURLSessionConfiguration `, but I can hook `__NSCFURLSessionConfiguration `, It is worked now. 

I fix this in  12dfbef and   cb67bf5.  It's a little diffrent with yours. You can modify this two file, I will release this in 1.1.8.


One more thing, I find the reason in your screenshot, I init a UIView in LLConfig's initial method, but you called [LLConfig sharedConfig] in a sub thread at first time.   You can called it first in mainThread, All database operation will be work in sub thread. I will alse fix this problem in 1..1.8.

Thank you very much indeed!
